src/mds/CDir.cc

```diff
@@ -1773,8 +1773,14 @@ void CDir::_omap_fetched(bufferlist& hdrbl, map<string, bufferlist>& omap,
 	}
       }
     } else {
-      dout(1) << "corrupt directory, i got tag char '" << type << "' pos " << pos << dendl;
-      assert(0);
+      dout(1) << "corrupt directory, i got tag char '" << type << "' pos "
+        << pos << dendl;
+      cache->mds->clog->error() << "Corrupt directory entry '" << p->first
+        << "' in dirfrag " << *this;
+      // TODO: add a mechanism for selectively marking a path
+      // damaged, rather than marking the whole rank damaged.
+      cache->mds->damaged();
+      assert(0);  // Unreachable: damaged() respawns us
```

Comment:
At the moment this is a lesser of two evils situation: choosing between asserting out and damaged().  Currently, the damage to the inode is specific to the rank, because we can never migrate something to another rank without being able to load it.

Ideas or plans: sure :-)  This will need to be a compatible mechanism with the forward scrub, as a per-subtree repository of where we have seen damage.  Doesn't necessarily need to be persistent initially, as we can always re-discover that a particular path is unreadable.